### PR TITLE
[BB-840] Include the optional unbound role

### PR DESCRIPTION
**Pull request by lgp171188**
This will be used by haproxy-stage.

### CODE REVIEW

**Comment by UmanShahzad** [comment]
@lgp171188 This doesn't seem ready for review as the tests are failing for related reasons: https://circleci.com/gh/open-craft/ansible-load-balancer/208?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: defaults/main.yml [hunk 1]
```diff
@@ -66,3 +66,6 @@ haproxy_exporter_consul_service:
       - monitor-https
     port: 19101
     enable_tag_override: true
+
+# Unbound #####################################################################
+caching_resolver_enabled: false
```
Comment (UmanShahzad) [inline defaults/main.yml:71]: Nit: 1 newline above this

File: meta/main.yml [hunk 2]
```diff
@@ -12,3 +12,6 @@ dependencies:
     NGINX_PROXY_CONFIG_PATH: /etc/nginx/sites-available/haproxy-exporter
     NGINX_PROXY_PUBLIC_PORT: 19101
     NGINX_PROXY_INTERNAL_PORT: 9101
+  - name: unbound
+    tags: 'unbound'
+    when: caching_resolver_enabled
```
Comment (UmanShahzad) [inline meta/main.yml:17]: Nit: put `tags` last since it's metadata

**Review by UmanShahzad — CHANGES_REQUESTED** [review]

**Comment by UmanShahzad** [comment]
Another thing: I think the configuration of unbound should probably go into ansible-secrets hosts, and the role should be included within a playbook rather than as a meta? It isn't strictly needed to run a load balancer, so it isn't a 'dependency'.

**Comment by lgp171188** [comment]
> This doesn't seem ready for review as the tests are failing for related reasons: https://circleci.com/gh/open-craft/ansible-load-balancer/208?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

@UmanShahzad, I didn't know that this repository has CI checks :open_mouth: I checked the error and it looks like it is related to the CI checks not using the related `ansible-playbooks` PR branch. I can look into it, however that may not be required due to the below comment.

> Another thing: I think the configuration of unbound should probably go into ansible-secrets hosts, and the role should be included within a playbook rather than as a meta? It isn't strictly needed to run a load balancer, so it isn't a 'dependency'.

It makes sense to move this to the `ansible-playbooks` repository :+1: So, if you don't have any objections, I will close this PR and move the changes to the load balancer playbook in the other PR.

**Comment by lgp171188** [comment]
Closing this in favor of https://github.com/open-craft/ansible-playbooks/pull/105 which has the changes to conditionally install unbound.

**Pull request closed without merging**